.github/workflows/test-external.yml

```
@@ -22,43 +24,59 @@ jobs:
         project:
           - repo: "morpho-org/morpho-data-structures"
             dir: "morpho-data-structures"
-            cmd: "halmos --function testProve --loop 4 --symbolic-storage"
+            cmd: "--function testProve --loop 4 --symbolic-storage"
             branch: ""
             profile: ""
           - repo: "a16z/cicada"
             dir: "cicada"
-            cmd: "halmos --contract LibUint1024Test --function testProve --loop 256"
+            cmd: "--contract LibUint1024Test --function testProve --loop 256 --solver-command yices-smt2"
             branch: ""
             profile: ""
           - repo: "a16z/cicada"
             dir: "cicada"
-            cmd: "halmos --contract LibPrimeTest --function testProve --loop 256"
+            cmd: "--contract LibPrimeTest --function testProve --loop 256 --solver-command bitwuzla"
             branch: ""
             profile: ""
           - repo: "farcasterxyz/contracts"
             dir: "farcaster-contracts"
-            cmd: "halmos"
+            cmd: "--solver-command yices-smt2"
             branch: ""
             profile: ""
           - repo: "zobront/halmos-solady"
             dir: "halmos-solady"
-            cmd: "halmos --function testCheck"
+            cmd: "--function testCheck --solver-command yices-smt2"
             branch: ""
             profile: ""
           - repo: "pcaversaccio/snekmate"
             dir: "snekmate"
-            cmd: "halmos --config test/halmos.toml"
+            cmd: "--config test/halmos.toml --solver-command yices-smt2"
             branch: ""
             profile: "halmos"
 
     steps:
-      - name: Checkout
+      # TODO: remove this step once halmos-builder package is public
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Checkout halmos
         uses: actions/checkout@v4
         with:
-          path: halmos
           # we won't be needing tests/lib for this workflow
           submodules: false
 
+      - name: Build image
+        run: docker build -t halmos-image . --file packages/halmos/Dockerfile
+
+      - name: Install Vyper
+        if: ${{ matrix.project.dir == 'snekmate' }}
+        run: |
+          docker run --name halmos-vyper --entrypoint uv halmos-image pip install vyper
```

would this work as well?  having `halmos-image` placed between `uv` and `pip install vyper` is a bit confusing.  wdyt?
```suggestion
          docker run --name halmos-vyper halmos-image --entrypoint uv pip install vyper
```

---

no, it's a weird docker command line quirk. I agree it's confusing but I couldn't find a workaround, it really insists on this format

`Usage:  docker run [OPTIONS] IMAGE [COMMAND] [ARG...]`

---

if you invoke `docker run --name halmos-vyper halmos-image --entrypoint uv pip install vyper`, it will invoke the current entrypoint of `halmos-image` with the rest of the arguments, so it will run `halmos --entrypoint uv pip install vyper`

---

oh i see, TIL!